zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -327,4 +319,53 @@ public static ResponseEntity<Object> mapDocumentHandlingExceptionToResponse(
       final DocumentException e) {
     return mapProblemToResponse(mapDocumentHandlingExceptionToProblem(e));
   }
+
+  public static ProblemDetail mapCamundaSearchExceptionToProblem(
+      CamundaSearchException cse, final Function<BrokerRejection, ProblemDetail> rejectionMapper) {
+    final CamundaSearchException.Reason reason =
+        (cse.getReason() == null) ? CamundaSearchException.Reason.UNKNOWN : cse.getReason();
+    final Throwable cause = (cse.getCause() == null) ? cse : cse.getCause();
+    final String name = cause.getClass().getName();
+    final String message = cause.getMessage();
```

Comment:
🔧 We should only need the cause's message in case of an UNKNOWN reason. Otherwise, the message should be the one from the `CamundaSearchException`. In the logger messages, we can go for the "Expected to handle ..., but ..." texts. Those should NOT go into the user-facing problem details though. They already know they sent a REST request. But in the logs it's helpful.